v7.3.2 (#12971)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.3.2

_May 2, 2024_

> We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:

Reviewer: ```suggestion
We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
```

- 🎁 Add "no data" and "loading" states to charts, allowing users to create [custom visualizations for each state](https://mui.com/x/react-charts/styling/#overlay)

> - 📝 Fixed a typo in an undocumented [x-charts gradient id](https://github.com/mui/mui-x/pull/12941/files/4bf4bffbc2799a01a96bc7458a17318cf41c1722#diff-26c31cc69d6f51110f89e339578ef9b3d4a3551f79077fff73f7babb81c5099f) that could have unintended side effects

Reviewer 2: Nitpick: Is it worth adding such a thing into highlights? 🤔

Author: I was unsure where to put it, moved it to just below the commit

Reviewer 2: Oh, sorry, I didn't check what is all it about. 🙈 
Yeah, in that case, I think that mentioning it in highlights makes much more sense...
Sorry for the confusion. 🤷 
I'd maybe rephrase it to be more specific, targetting the actual problem - change in the id of an element instead of saying "unintended side effects", which sounds more scary to me, when reading about it. 🙈

Author: done

> - 🌍 Improve Catalan (he-IL) and Spanish (hu-HU) locales on the Date and Time Pickers

Reviewer: ```suggestion
- 🌍 Improve Hebrew (he-IL) and Hungarian (hu-HU) locales on the Date and Time Pickers
```

> - 🌍 Improve Catalan (he-IL) and Spanish (hu-HU) locales on the Date and Time Pickers

Reviewer 2: ```suggestion
- 🌍 Improve Hebrew (he-IL) and Hungarian (hu-HU) locales on the Date and Time Pickers
```

> - 🐞 Bugfixes

Reviewer: ```suggestion
- 🌍 Improve Slovak (sk-SK) locale on the Data Grid
- 🐞 Bugfixes
```

- 📚 Documentation improvements

### Data Grid

#### `@mui/x-data-grid@7.3.2`

> 

Reviewer: ```suggestion

[l10n] Improve Slovak (sk-SK) locale (#12949) @stefikp
```

- [DataGrid] Derive `formattedValue` from the edit value when passing to `renderEditCell` (#12870) @cherniavskii

> 

Reviewer: ```suggestion
- [DataGrid] Fix rows not being recomputed on `props.rowCount` change (#12833) @MBilalShafi

```

#### `@mui/x-data-grid-pro@7.3.2` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.3.2`.

#### `@mui/x-data-grid-premium@7.3.2` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

> Same changes as in `@mui/x-data-grid-pro@7.3.2`.

Reviewer: ```suggestion
Same changes as in `@mui/x-data-grid-pro@7.3.2`, plus:

- [DataGridPremium] Fix print export not working with row grouping (#12957) @MBilalShafi

```

### Date and Time Pickers

#### `@mui/x-date-pickers@7.3.2`

- [l10n] Improve Hebrew (he-IL) locale (#12910) @michaelNXT1
- [l10n] Improve Hungarian (hu-HU) locale (#12930) @noherczeg
- [pickers] Fix typo on the `viewRenderers` prop description (#12915) @flaviendelangle
- [pickers] Improve TypeScript performance in `PickersDay` (#12920) @flaviendelangle

#### `@mui/x-date-pickers-pro@7.3.2` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.3.2`.

### Charts

#### `@mui/x-charts@7.3.2`

- [charts] Add an overlay for "no data" or "loading" states (#12817) @alexfauquette
- [charts] Fix typos in documentation, translations and errors (#12941) @JCQuintas
- [charts] Fix `prop.slots` and `prop.slotProps` not passed to `<ChartsTooltip />` (#12939) @JCQuintas

### Docs

> - [docs] Polish references to the plans (#12922) @oliviertassinari

Reviewer: ```suggestion
- [docs] Improve Data Grid migration guide (#12969) @MBilalShafi
- [docs] Polish references to the plans (#12922) @oliviertassinari
```

Same changes as in `@mui/x-data-grid@7.3.2`.

#### `@mui/x-data-grid-premium@7.3.2` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.3.2`.

### Date and Time Pickers

#### `@mui/x-date-pickers@7.3.2`

- [l10n] Improve Hebrew (he-IL) locale (#12910) @michaelNXT1
- [l10n] Improve Hungarian (hu-HU) locale (#12930) @noherczeg
- [pickers] Fix typo on the `viewRenderers` prop description (#12915) @flaviendelangle
- [pickers] Improve TypeScript performance in `PickersDay` (#12920) @flaviendelangle

#### `@mui/x-date-pickers-pro@7.3.2` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@7.3.2`.

### Charts

#### `@mui/x-charts@7.3.2`

- [charts] Add an overlay for "no data" or "loading" states (#12817) @alexfauquette
- [charts] Fix typos in documentation, translations and errors (#12941) @JCQuintas
- [charts] Fix `prop.slots` and `prop.slotProps` not passed to `<ChartsTooltip />` (#12939) @JCQuintas

### Tree View

> #### `@mui/x-tree-view@7.3.2`

Author: Do we bump when there are no changes?

Reviewer 2: Lately we've been releasing whatever lerna detects, that sometimes means that packages, which only received dependency bumps would also be picked up.
This decision has not been finalized yet, but we did release such packages with the last few releases.
Personally, I lean more towards not releasing what has not changed, but that's just my opinion with "hyper-optimisation" in mind. 😆

Author: I removed it for now as @michelengelen suggested. I don't have any preference, but would like to add our decision to the readme file so it is documented for the next new joiner 😅

Reviewer 2: I'm fine with it.
Totally agree that it should be documented once we land on a decision. 👌

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.3.2

_May 2, 2024_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

- 🎁 Add "no data" and "loading" states to charts, allowing users to create [custom visualizations for each state](https://mui.com/x/react-charts/styling/#overlay)
- 🌍 Improve Hebrew (he-IL) and Hungarian (hu-HU) locales on the Date and Time Pickers
- 🌍 Improve Slovak (sk-SK) locale on the Data Grid
- 🐞 Bugfixes
- 📚 Documentation improvements

### Data Grid

#### `@mui/x-data-grid@7.3.2`

> [l10n] Improve Slovak (sk-SK) locale (#12949) @stefikp

Reviewer: ```suggestion
```

- [DataGrid] Derive `formattedValue` from the edit value when passing to `renderEditCell` (#12870) @cherniavskii
- [DataGrid] Fix rows not being recomputed on `props.rowCount` change (#12833) @MBilalShafi

> 

Reviewer: ```suggestion
- [l10n] Improve Slovak (sk-SK) locale (#12949) @stefikp

```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.3.2

_May 2, 2024_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

- 🎁 Add "no data" and "loading" states to charts, allowing users to create [custom visualizations for each state](https://mui.com/x/react-charts/styling/#overlay)
- 🌍 Improve Hebrew (he-IL) and Hungarian (hu-HU) locales on the Date and Time Pickers
- 🌍 Improve Danish (da-DK) and Slovak (sk-SK) locales on the Data Grid
- 📝 Fix a [typo](https://github.com/mui/mui-x/pull/12941/files/4bf4bffbc2799a01a96bc7458a17318cf41c1722#diff-26c31cc69d6f51110f89e339578ef9b3d4a3551f79077fff73f7babb81c5099f) in the auto-generated Charts gradient `id` attribute.
  It should not affect you, but if you were relying on the gradient `id` attribute, please update your usage.
- 🐞 Bugfixes
- 📚 Documentation improvements

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@7.3.2`

> - [DataGrid] Derive `formattedValue` from the edit value when passing to `renderEditCell` (#12870) @cherniavskii

Reviewer: Could we add this one too please?
```suggestion
- [DataGrid] Allow to change reset text in columns management panel (#12972) @MBilalShafi
- [DataGrid] Derive `formattedValue` from the edit value when passing to `renderEditCell` (#12870) @cherniavskii
```

- [charts] Add an overlay for "no data" or "loading" states (#12817) @alexfauquette
- [charts] Fix typos in documentation, translations and errors (#12941) @JCQuintas

>   This affects an undocumented [x-charts gradient id](https://github.com/mui/mui-x/pull/12941/files/4bf4bffbc2799a01a96bc7458a17318cf41c1722#diff-26c31cc69d6f51110f89e339578ef9b3d4a3551f79077fff73f7babb81c5099f) that could have unintended side effects

Reviewer: Should we add this under the breaking changes section?

Author: Technically a breaking change yes, but it is also very small. @LukasTy suggested just adding it to the changelog. Do you think this should be in a `### Breaking change` section?

Reviewer 2: IMHO, it's such a small BC that flagging it as one might be a bit counter-intuitive and cause more confusion than necessary. 🙈

Reviewer: Go ahead with what you feel better, IMHO, putting in the breaking changes section would make more sense, as we used to do the same for all other BCs.
